mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala

```diff
@@ -179,7 +179,7 @@ abstract class GeneralizedLinearAlgorithm[M <: GeneralizedLinearModel]
    */
   def run(input: RDD[LabeledPoint]): M = {
     if (numFeatures < 0) {
-      numFeatures = input.map(_.features.size).first()
+      numFeatures = input.first().features.size
```

Comment:
I see your intent here, but if the vectors are very large, it might actually be faster to get its size remotely and return only the size to the driver.  But yes it entails a small distributed operation. I'm not sure which is better.